### PR TITLE
Remove Hardcoded Items and Implement Dynamic Task Management Functionality

### PR DESCRIPTION
**Pull request by kit0-0**
**This pull request removes hardcoded items from the tasks array, introduces a new JavaScript file for enhanced functionality, and implements several key features:**

- Removed hardcoded items
- Created a new JavaScript file, index.js, specifically dedicated to the new functionality
- Implemented a function that enables the addition of new tasks to the array. Each new task has the property completed set to false and the property index set to the value of the new array length
- Delete Task Functionality: double click on list input can see the functionality to  delete
- Edit Task Functionality
- All changes made to the To Do List, including task additions, deletions, and edits, are now saved in the local storage

- [Live Demo Link](https://kit0-0.github.io/To-Do-list/)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,42 +1,134 @@
...
+      storeTasksToLocalStorage();
+      displayTasks();
     });
+  };
+
+  const editTask = (description, index) => {
+    tasks[index].description = description;
+    storeTasksToLocalStorage();
+  };
+
```
Comment (julie-ify) [inline src/index.js]: - [ ] The delete and edit features should be implemented in a new file as stated by the guidelines. Kindly do move these methods to a new file

File: src/index.js [hunk 2]
```diff
@@ -1,42 +1,134 @@
...
+      description: taskDescription,
+      completed: false,
+      index: tasks.length + 1, // Set the index as array length + 1
+    };
+    tasks.push(newTask);
+    storeTasksToLocalStorage();
+    inputField.value = '';
+    displayTasks();
+  }
+};
```
Comment (julie-ify) [inline src/index.js]: - [ ] The add task feature should be a new file also. Kindly move this method. You can add to the file with the remove and edit methods or create a new file for it, but it just should not be here in the `index.js` file for better code organization.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @kit0-0,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kit0-0** [comment]
**Changes Made**

- Moved addTask, deleteTask, and editTask functions to taskFunctions.js

**Review by omarsalem7 — APPROVED** [review]
Hi @kit0-0 😊, 
# STATUS: APPROVED! 🥇 💯 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**